src/NHibernate/Multi/MultiAnyQueryBase.cs

```diff
@@ -0,0 +1,217 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Linq;
+using NHibernate.Cache;
+using NHibernate.Engine;
+using NHibernate.SqlCommand;
+using NHibernate.Transform;
+using NHibernate.Util;
+
+namespace NHibernate
+{
+	/// <summary>
+	/// Base class for both ICriteria and IQuery queries
+	/// </summary>
+	public abstract partial class MultiAnyQueryBase<TResult> : IMultiAnyQuery, IMultiAnyQuery<TResult>
```

Comment:
`IMultiAnyQuery` already comes with `IMultiAnyQuery<TResult>`, reduce redundancy. An same remark than previously for naming.